--- rcs-5.8/man/rcsfile.5in.orig	2011-01-09 17:00:46.000000000 +0100
+++ rcs-5.8/man/rcsfile.5in	2011-09-11 20:56:42.648245946 +0200
@@ -119,6 +119,18 @@
 Dates use the Gregorian calendar; times use UTC.
 .PP
 The
+.I commitid
+is followed by an
+.I id
+token. This token is intended to be unique across
+multiple files and is used to help group files as
+being a part of the same logical commit.
+This token must uniquely identify the commit
+operation that was applied to a set of RCS files.
+In particular, it must be unique among all the
+commitids in this file.
+.PP
+The
 .I delta
 nodes form a tree.  All nodes whose numbers
 consist of a single pair
--- rcs-5.8/src/ident.c.orig	2011-01-07 13:27:18.000000000 +0100
+++ rcs-5.8/src/ident.c	2011-09-11 20:55:08.061576114 +0200
@@ -51,6 +51,7 @@
         {
         case LETTER:
         case Letter:
+        case DIGIT:
           *tp++ = c;
           if (tp < line + sizeof (line) - 4)
             break;
