diff -Nru rcs-5.7/man/Makefile.in rcs-5.7.new/man/Makefile.in
--- rcs-5.7/man/Makefile.in	Fri Jun 16 08:19:24 1995
+++ rcs-5.7.new/man/Makefile.in	Fri Jun 25 17:00:07 1999
@@ -34,8 +34,8 @@
 PIC = @PIC@
 
 prefix = @prefix@
-man1dir = $(prefix)/man/man1
-man5dir = $(prefix)/man/man5
+man1dir = @mandir@/man1
+man5dir = @mandir@/man5
 man1ext = .1
 man5ext = .5
 
@@ -49,16 +49,16 @@
 check dvi info TAGS ::
 
 installdirs :: ../mkinstalldirs
-	-$(srcdir)/../mkinstalldirs $(man1dir) $(man5dir)
+	-$(srcdir)/../mkinstalldirs $(DESTDIR)$(man1dir) $(DESTDIR)$(man5dir)
 
 man1pages = ci co ident merge rcs rcsclean rcsdiff rcsintro rcsmerge rlog
 
 install :: installdirs
 	-for m in $(man1pages); do \
-		$(INSTALL_DATA) $(srcdir)/$$m.1 $(man1dir)/$$m$(man1ext); \
+		$(INSTALL_DATA) $(srcdir)/$$m.1 $(DESTDIR)$(man1dir)/$$m$(man1ext); \
 	done
 	-{ test -f rcsfile.5 || cd $(srcdir); } && \
-	$(INSTALL_DATA) rcsfile.5 $(man5dir)/rcsfile$(man5ext)
+	$(INSTALL_DATA) rcsfile.5 $(DESTDIR)$(man5dir)/rcsfile$(man5ext)
 
 uninstall ::
 	for m in $(man1pages); do \
diff -Nru rcs-5.7/src/Makefile.in rcs-5.7.new/src/Makefile.in
--- rcs-5.7/src/Makefile.in	Fri Jun 16 08:19:24 1995
+++ rcs-5.7.new/src/Makefile.in	Fri Jun 25 17:00:52 1999
@@ -65,7 +65,7 @@
 # The following definitions can be tailored by hand;
 # this shouldn't be needed for ordinary installations.
 
- bindir = $(exec_prefix)/bin
+ bindir = @bindir@
 
  ALL_CFLAGS = $(CPPFLAGS) -Dhas_conf_h $(DEFS) -I. -I$(srcdir) $(CFLAGS)
 
@@ -110,11 +110,11 @@
 	$(CC) -c $(ALL_CFLAGS) $<
 
 installdirs :: ../mkinstalldirs
-	$(srcdir)/../mkinstalldirs $(bindir)
+	$(srcdir)/../mkinstalldirs $(DESTDIR)$(bindir)
 
 install :: all installdirs
 	for p in $(PROGRAMS); do \
-	  $(INSTALL_PROGRAM) $$p $(bindir)/$$p; \
+	  $(INSTALL_PROGRAM) $$p $(DESTDIR)$(bindir)/$$p; \
 	done
 
 uninstall ::
